app/controllers/thread_controller.php

```diff
@@ -80,4 +97,101 @@ public function create()
         $this->set(get_defined_vars());
         $this->render($page);
     }
+
+    public function edit()
+    {
+        $check = Param::get('check', false);
+        $title = " | Edit thread";
+        $user = new User($_SESSION['username']);
+        $params = array(
+            'thread_id' => Param::get('id', 0),
+            'user_id'   => $user->user_id,
+        );
+        $thread = new Thread($params);
+
+        $thread_content = Thread::get(Param::get('id', 0));
+        $thread->error = isset($thread_content->error) ? $thread_content->error : null;
+        if (isset($thread->error)) {
+            $this->set(get_defined_vars());
+            return;
+        }
+        if ($thread_content->user_id !== $user->user_id) {
+            $thread->error = 'Cannot edit other user\'s thread';
+            $this->set(get_defined_vars());
+            return;
+        }
+        if ($check) {
+            $thread->title = Param::get('new_thread', '');
+            try {
+                $thread->editThread();
+            } catch (ValidationException $e) {
+                $thread->error = 'Input Error, Please enter at from 1 to 200 charcters';
+            } catch (Exception $e) {
+                $thread->error = 'Unexpected Error occured';
+            }
+        }
+        $this->set(get_defined_vars());
+    }
+
+    public function top_threads()
+    {
+        $type = Param::get('type', '');
+        if ($type === '') {
+            redirect(url('/'));
+        }
+        switch ($type) {
+            case 'comment':
+                $title = 'Most Commented Thread';
+                $sub_title = 'Showing top %d most commented threads';
+                break;
+            case 'follow':
+                $title = 'Most Followed Thread';
+                $sub_title = 'Showing top %d most followed threads';
+                break;
+            default:
+                break;
+        }
+        $user = new User($_SESSION['username']);
+        $thread = new Thread;
+        $top_threads = $thread->top($type);
```

Comment:
ah yes. its much better if you use the first one. because models are for data objects in your database or table. so, if you have the model `Thread`, it must perform data manipulation and crud operations for `thread` table only. :bow: